### ISSUE TITLE
Guidance on package archiving

### ISSUE THREAD

**Issue by maelle** [ropensci/dev_guide#156]
@noamross, you wrote "Archiving - eventually making sure all RO packages go to a scientific repository, not just JOSS packages." in https://discuss.ropensci.org/t/should-the-ropensci-dev-guide-include-package-citation-best-practices/1551/4

How should our guidance about that look like? Recommend setting up Zenodo integration + adding last DOI badge? There's no helper for that right?

And then how to add that latest DOI in the CITATION file automatically? cf also #115

**Comment by maelle** [comment]
Note that our approval issue template has an item "Activate Zenodo watching the repo".

**Comment by maelle** [comment]
In the current approval template there's

```
<IF JOSS>
- [ ] Activate Zenodo watching the repo
- [ ] Tag and create a release so as to create a Zenodo version and DOI
- [ ] Submit to JOSS using the Zenodo DOI.  We will tag it for expedited review.
<IF JOSS/>
```

We could make it 

```
- [ ] Activate Zenodo watching the repo
- [ ] Tag and create a release so as to create a Zenodo version and DOI
- [ ] Add a latest DOI badge to your README and create a CITATION file blablabla.
<IF JOSS>
- [ ] Submit to JOSS using the Zenodo DOI.  We will tag it for expedited review.
<IF JOSS/>
```

It'd be actually a bit complicated if we were to recommend something else than Zenodo for archiving: some repos would end up with two types of archiving if they were submitted to JOSS.

**Comment by noamross** [comment]
This seems fine, but I think a better way to in the medium term would be for us to do this, preferably in an automated way.  Like, after repo transfer, a single post-approval button should make Zenodo archiving happen, build the docs, etc.  (Relatedly we could run some sort of check to make sure CRAN releases after acceptance are also tagged/archived.)

**Comment by maelle** [comment]
The docs are now built centrally, for each push after repo transfer (very new system though).

Wouldn't the authors want to own the Zenodo thing? And even if we made the archiving happen, it'd be on the authors to add a badge?

**Comment by noamross** [comment]
Yes, authors should control their own Zenodo record.  I just hate piling more tasks on rather than automating them away.  Does Zenodo read codemeta.json files yet?  If not, maybe **rodev** can import **codemetar** and generate both the codemeta and .zenodo.json files? Could it then also insert the badge and make the citation file?

**Comment by maelle** [comment]
I don't think Zenodo reads Codemeta yet cc @cboettig 

Even if creating a .zenodo.json there'd be something to activate over at Zenodo, correct?

Automatically inserting the badge will only work fully automatically if the README has markers for the badges paragraph but the function could use `usethis::use_badge()` that gives an informative message when the badge needs to be added by hand. How would one know what ID to have in the badge? I guess that it's related to my question in the previous paragraph + the R tooling for Zenodo API. :thinking:

**Comment by maelle** [comment]
Postponing this until we get/make an up-to-date Zenodo R client.

**Comment by maelle** [comment]
To be explored https://cran.r-project.org/web/packages/zen4R/index.html

**Comment by maelle** [comment]
R package + PR bot?

**Comment by maelle** [comment]
The topic of package archiving is related to the topic of package citation.

cc @mpadge

**Comment by mpadge** [comment]
I think that this issue is best postponed for now and merged with development of [`ropenscilabs/deposits`](https://github.com/ropenscilabs/deposits). I'll open a parallel issue there for a single function which authors can then call locally to automatically create the zenodo deposit. Until then, milestone will be bumped to 0.9.

**Comment by maelle** [comment]
@mpadge could your current work on deposits help with this or should we shift this to a later milestone?

**Comment by mpadge** [comment]
Yes, please shift to a later milestone. Thanks

**Comment by maelle** [comment]
@mpadge any update on this?

**Comment by maelle** [comment]
@mpadge curious to hear whether deposits in its current state can help with this :smile_cat:

**Comment by mpadge** [comment]
Yes, it can indeed. I'm already assigned here, and will get on to it when i allocate dedicated DevGuide time. In September sometime ...
